### PR TITLE
Fix ingesters with less tokens stuck in LEAVING

### PR DESCRIPTION
**Pull request by marianafranco**
Signed-off-by: Mariana Franco <marfram@amazon.com>

**What this PR does**:  

It's possible to have ingesters in the ring with less tokens than configured because of conflict resolution:

https://github.com/cortexproject/cortex/blob/f930c7a82b7ba7a3b3be8607715a4396ec446746/pkg/ring/model.go#L354-L361

In this case, the ingester will get stuck in LEAVING when they are evicted and rescheduled.

Looking at previous changes, seems that there was not a good reason to check for NumTokens: https://github.com/cortexproject/cortex/pull/1921#discussion_r359392179 

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [X] Tests updated
- [ ] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by friedrichg — COMMENTED** [review]
Thank you for your contribution. 

I also notice that function is barely tested

Screen Shot 2022-12-27 at 11 30 10 AM

**Comment by marianafranco** [comment]
> I also notice that function is not completely tested with unit tests

Sorry, I don't have time to improve these unit tests now. I can create an issue if you want. Hopefully other contributors can help to improve the coverage.

File: pkg/ring/lifecycler_test.go [hunk 1]
```diff
@@ -755,10 +755,10 @@ func TestJoinInLeavingState(t *testing.T) {
 	defer services.StopAndAwaitTerminated(context.Background(), r) //nolint:errcheck
 
 	cfg := testLifecyclerConfig(ringConfig, "ing1")
-	cfg.NumTokens = 2
+	cfg.NumTokens = 3
```
[the remark this answers is not in the document]
Comment (marianafranco) [reply pkg/ring/lifecycler_test.go:811]: Done. I created a new test case.

**Review by friedrichg — APPROVED** [review]
Thanks!

**Comment by marianafranco** [comment]
@alanprot Could you take a look?

**Review by alanprot — APPROVED** [review]

**Pull request merged**